frontend/views/containers/contributions/Contribution.vue

```diff
@@ -142,21 +143,24 @@ export default {
       }
       return this.isFilled ? this.handleSubmit() : this.handleDelete()
     },
-    handleDelete () {
-      this.$emit('new-value', 'nonMonetaryRemove', this.initialValue)
+    async handleDelete () {
+      await this.$listeners['new-value']('nonMonetaryRemove', this.initialValue)
```

Comment:
One potential issue with this new way of calling the `new-value` handler is that it will cause an exception/error if the handler is not an asynchronous function.

So either you need to check for the existence of `this.$listeners['new-value'].then` before calling `await`, or you need to clearly name and document (with comments) this event as something that must be bound to an asynchronous function only.

Reply:
> it will cause an exception/error if the handler is not an asynchronous function

I just did the test by calling a normal function and everything worked correctly, even if it's a sync function.

Based on [this explanation](https://stackoverflow.com/a/55263084/4737729) there isn't any perceivable consequence.

Also, in the ButtonSumibt docs, this is the first sentence:

> Use ButtonSubmit on buttons that will trigger an **async** action.

Reply:
> Also, at ButtonSumibt.vue docs, this is the first sentence:

OK, but that has nothing to do with `Contribution.vue` and this `new-value` event?

> I just did the test by calling a normal function and everything worked correctly, even if it's a sync function.

Hmm.. that's odd. OK, I will test it out myself later today and if it works as you say then we can ignore this.

Reply:
`ButtonSubmit` calls the listener `@click`. That listener triggers `async handleDelete` which then awaits for the $listener `'new-value'`. That's the relationship between both.

Reply:
OK, I just verified that there's no issue with non-async handlers! 👍